driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -108,7 +109,7 @@ public interface Expression {
     IntegerExpression isIntegerOr(IntegerExpression other);
     StringExpression isStringOr(StringExpression other);
     DateExpression isDateOr(DateExpression other);
-    <T extends Expression> ArrayExpression<T> isArrayOr(ArrayExpression<? extends T> other);
+    <T extends Expression> ArrayExpression<@MqlUnchecked T> isArrayOr(ArrayExpression<? extends T> other);
```

Comment:
I like this.

I think we should have no default, and specify the issue each time. For example, here, it should perhaps be `MqlUnchecked(GENERIC)`? This way, the user might click on the term to get guidance.

Reply:
`MqlUnchecked(TYPE_ARGUMENT)` then I'd say, as the inferred/user-specified [type argument](https://docs.oracle.com/javase/specs/jls/se17/html/jls-4.html#jls-4.5.1) of the parameterized type returned by the method is exactly where the API gives up on checking.